lib/MusicBrainz/Server/Form/User/EditProfile.pm

```diff
@@ -7,9 +7,15 @@ use MusicBrainz::Server::Translation qw( l ln );
 use MusicBrainz::Server::Validation qw( is_valid_url );
 
 extends 'MusicBrainz::Server::Form';
+with 'MusicBrainz::Server::Form::Role::ValidateUsername';
 
 has '+name' => ( default => 'profile' );
 
+has_field 'username' => (
```

Comment:
We'll need a check for that in `Controller::Account`, I think. I would just `$c->detach('/error_403')` if the username changed and the user isn't an account admin.